controllers/openshift_controller.go

```diff
@@ -1041,32 +1041,177 @@ func (r *KataConfigOpenShiftReconciler) createExtensionMc(machinePool string) (c
 	return ctrl.Result{}, nil, false
 }
 
-func (r *KataConfigOpenShiftReconciler) mapKataConfigToRequests(kataConfigObj client.Object) []reconcile.Request {
+func (r *KataConfigOpenShiftReconciler) makeReconcileRequest() reconcile.Request {
+	return reconcile.Request{
+		NamespacedName: types.NamespacedName{
+			Name: r.kataConfig.Name,
+		},
+	}
+}
+
+func isMcpRelevant(mcp client.Object) bool {
+	mcpName := mcp.GetName()
+	if mcpName == "kata-oc" || mcpName == "worker" {
+		return true
+	}
+	return false
+}
 
-	kataConfigList := &kataconfigurationv1.KataConfigList{}
+func logMcpChange(log logr.Logger, statusOld, statusNew mcfgv1.MachineConfigPoolStatus) {
 
-	err := r.Client.List(context.TODO(), kataConfigList)
-	if err != nil {
-		return []reconcile.Request{}
+	log.Info("MCP updated")
+
+	if statusOld.MachineCount != statusNew.MachineCount {
+		log.Info("MachineCount changed", "old", statusOld.MachineCount, "new", statusNew.MachineCount)
+	} else {
+		log.Info("MachineCount", "#", statusNew.MachineCount)
+	}
+	if statusOld.ReadyMachineCount != statusNew.ReadyMachineCount {
+		log.Info("ReadyMachineCount changed", "old", statusOld.ReadyMachineCount, "new", statusNew.ReadyMachineCount)
+	} else {
+		log.Info("ReadyMachineCount", "#", statusNew.ReadyMachineCount)
+	}
+	if statusOld.UpdatedMachineCount != statusNew.UpdatedMachineCount {
+		log.Info("UpdatedMachineCount changed", "old", statusOld.UpdatedMachineCount, "new", statusNew.UpdatedMachineCount)
+	} else {
+		log.Info("UpdatedMachineCount", "#", statusNew.UpdatedMachineCount)
+	}
+	if statusOld.DegradedMachineCount != statusNew.DegradedMachineCount {
+		log.Info("DegradedMachineCount changed", "old", statusOld.DegradedMachineCount, "new", statusNew.DegradedMachineCount)
+	} else {
+		log.Info("DegradedMachineCount", "#", statusNew.DegradedMachineCount)
+	}
+	if statusOld.ObservedGeneration != statusNew.ObservedGeneration {
+		log.Info("ObservedGeneration changed", "old", statusOld.ObservedGeneration, "new", statusNew.ObservedGeneration)
 	}
 
-	reconcileRequests := make([]reconcile.Request, len(kataConfigList.Items))
-	for _, kataconfig := range kataConfigList.Items {
-		reconcileRequests = append(reconcileRequests, reconcile.Request{
-			NamespacedName: types.NamespacedName{
-				Name: kataconfig.Name,
-			},
-		})
+	if !reflect.DeepEqual(statusOld.Conditions, statusNew.Conditions) {
+
+		for _, condType := range []mcfgv1.MachineConfigPoolConditionType{"Updating", "Updated"} {
+			condOld := mcfgv1.GetMachineConfigPoolCondition(statusOld, condType)
+			condNew := mcfgv1.GetMachineConfigPoolCondition(statusNew, condType)
+			condStatusOld := "<missing>"
```

Comment:
Is `<missing>` status string coming from the mcp code ?

Reply:
Nope, it's just a string I picked because a) it's not a valid condition value and b) looks good (=is easy to understand) in the log.

It's meant to mean that the condition (`Updating` or `Updated`) is literally missing from the MCP's condition array - which it legally can.

Reply:
Probably better to add the above as a comment in the code to help us when we look at the code after few months :-)

Reply:
I'm not against but I can't think of a wording that said anything that wasn't in the code already.  I mean, the function is called `logMcpChange()` so it seems expected that it just composes log messages...

Reply:
@pmores I should have worded it better. My suggestion was specific to the `<missing>` status string. 
Maybe you can declare a constant like `const missingStatusString ..` and use it. When we look back at this code after a few months, we'll know what `<missing>` is for and not get confused about why we are using `<missing>` and not an empty string.
Btw just to be clear I'm fine with the code as-is. It was just a suggestion based on past experiences.

Reply:
Oh I see, I'm sorry, I misunderstood. Will do!